### PR TITLE
Migrate JMX Metric Extension to JMX Receiver

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
**Description:**

Per discussion in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1182 and some offline planning w/ @tigrannajaryan and @jrcamp, these proposed changes relocate the jmx metric extension to a new receiver.  This decision was arrived at by the JMX Metric Gatherer's initial use case likely always being associated with a single pipeline.  Requiring a separate extension with sole purpose of managing a JRE subprocess is limited in light of this and poor experience overall, especially when coupled w/ missing Observer compatibility for dynamic service urls.

These changes don't include any major functional updates, and a subsequent PR will include an implicit but configurable OTLP receiver.

**Testing:**

Existing tests updated as necessary.

**Documentation:**

Existing documentation updated as necessary.

### CODE REVIEW

File: receiver/jmxmetricreceiver/README.md [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
 
 Example configuration:
 
 ```yaml
-extensions:
+receivers:
   jmx_metrics:
     jar_path: /opt/opentelemetry-java-contrib-jmx-metrics.jar
     service_url: service:jmx:rmi:///jndi/rmi://<my-jmx-host>:<my-jmx-port>/jmxrmi
-    groovy_script: /opt/my/groovy.script
```
Comment (nilebox) [inline receiver/jmxmetricreceiver/README.md:30]: nit: why this change?

File: receiver/jmxmetricreceiver/config_test.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package jmxmetricextension
+package jmxmetricreceiver
```
Comment (nilebox) [inline receiver/jmxmetricreceiver/config_test.go:15]: nit: based on `jmx_metrics` name in the config, the package should be called `jmxmetricsreceiver` (i.e. plural).
Alternatively, we could just remove "metrics" completely and rename it to `jmx` in the config with a `jmxreceiver` package.

File: receiver/jmxmetricreceiver/integration_test.go [hunk 3]
```diff
@@ -234,14 +234,13 @@ func (suite *JMXIntegrationSuite) TestJMXMetricViaOTLPReceiverIntegration() {
 		Password:     "cassandra",
 	}
 
-	extension := newJMXMetricExtension(logger, config)
-	require.NotNil(t, extension)
+	receiver := newJMXMetricReceiver(logger, config, otlpReceiver)
```
Comment (nilebox) [inline receiver/jmxmetricreceiver/integration_test.go:237]: This is confusing, why next consumer is `otlpReceiver` here?
It should be either exporter or a processor I think?

Comment (nilebox) [reply receiver/jmxmetricreceiver/integration_test.go:237]: I see a similar test with Prometheus receiver below. So how does the "cascade" receiver combination work?

Comment (rmfitzpatrick) [reply receiver/jmxmetricreceiver/integration_test.go]: This was an unnecessary placeholder, and I've updated to use nil.  The complete receiver functionality isn't implemented yet, but when finished there will be an implicit otlp receiver created that receives the next consumer.

File: receiver/jmxmetricreceiver/README.md (hunk 1 above)
Comment (rmfitzpatrick) [reply receiver/jmxmetricreceiver/README.md]: This was to better advertise the built in target system groovy scripts, but since it's not related to the change I'll table it.

File: receiver/jmxmetricreceiver/config_test.go (hunk 2 above)
Comment (rmfitzpatrick) [reply receiver/jmxmetricreceiver/config_test.go:15]: @jrcamp @tigrannajaryan @bogdandrutu, thoughts?  I'm ok with `jmxreceiver`.

Comment (bogdandrutu) [reply receiver/jmxmetricreceiver/config_test.go:15]: I am fine with jmxreceiver, this is not very important :)

Comment (rmfitzpatrick) [reply receiver/jmxmetricreceiver/config_test.go:15]: Thanks, naming is always important imo, though I can get caught up in details...

Comment (tigrannajaryan) [reply receiver/jmxmetricreceiver/config_test.go]: Either `jmxmetricreceiver` or `jmxreceiver` SGTM, no preference, choose what you like most.

**Review by tigrannajaryan — APPROVED** [review]
LGTM
